manifests/params.pp

```diff
@@ -349,7 +349,9 @@
       $php_version = $facts['os']['release']['major'] ? {
         '9'     => '7.0', # Debian Stretch
         '10'    => '7.3', # Debian Buster
+        '16.04' => '7.0', # Ubuntu Xenial Xerus
         '20.04' => '7.4', # Ubuntu Foccal Fossal
+        '22.04' => '8.1', # Ubuntu Jellyfish
```

Comment:
I'd imagine 22.04 is closer to Debian Bullseye. It's mostly about the `mod_packages` list. So I'd suggest to change line 348 with a versioncmp. Possibly Ubuntu 20.04 is also more like Debian Bullseye than this list.

Reply:
I admit I don't know exactly what the `versioncmp()` does here. I would use the following block code instead (from L348 to L355:

```
if ($facts['os']['family'] == 'Debian') {
  $php_version = $facts['os']['release']['major'] ? {
    '9'      => '7.0',  # Debian Stretch
    '10'     => '7.3',  # Debian Buster
    '11'     => '7.4',  # Debian Bullseye
    '18.04'  => '7.2',  # Ubuntu Bionic
    '20.04'  => '7.4',  # Ubuntu Focal
    '22.04'  => '8.1',  # Ubuntu Jammy
    default: => '7.4',  # Debian Bullseye, Ubuntu Focal
  }
}
```

Reply:
I agree the `$php_version` part should be pulled out of the `versioncmp` and I swear I considered the exact same thing but failed to write it down. However, the list of modules can use a versioncmp. It's still not entirely correct, like https://github.com/puppetlabs/puppetlabs-apache/pull/2281 should clean things up further.